Examples/Io/Performance/ActsExamples/Io/Performance/VertexPerformanceWriter.cpp

```diff
@@ -291,51 +306,53 @@ ActsExamples::ProcessCode ActsExamples::VertexPerformanceWriter::writeT(
         }
 
         const auto& majorityParticle = *it;
-        associatedTruthParticles.emplace_hint(associatedTruthParticles.end(),
-                                              majorityParticle);
+        trackParameters.push_back(trajectories.trackParameters(tip));
+        associatedTruthParticles.push_back(majorityParticle);
       }
     }
   }
 
-  ACTS_VERBOSE(
-      "Total number of reconstructed tracks : " << trackParameters.size());
-
   // Get number of track-associated true primary vertices
   m_nVtxReconstructable =
-      getNumberOfReconstructableVertices(associatedTruthParticles);
+      getNumberOfReconstructableVertices(SimParticleContainer(
+          associatedTruthParticles.begin(), associatedTruthParticles.end()));
 
+  ACTS_INFO(
+      "Total number of reconstructed tracks : " << trackParameters.size());
   ACTS_INFO("Total number of reco track-associated truth particles in event : "
             << associatedTruthParticles.size());
   ACTS_INFO("Total number of reco track-associated truth primary vertices : "
             << m_nVtxReconstructable);
 
   // Loop over all reco vertices and find associated truth particles
   std::vector<SimParticleContainer> truthParticlesAtVtxContainer;
+
   for (const auto& vtx : vertices) {
     const auto tracks = vtx.tracks();
+
     // Store all associated truth particles to current vtx
     SimParticleContainer particleAtVtx;
-
     std::vector<int> contributingTruthVertices;
 
     for (const auto& trk : tracks) {
       Acts::BoundTrackParameters origTrack = *(trk.originalParams);
 
       // Find associated truth particle now
-      int idx = 0;
-      for (const auto& particle : associatedTruthParticles) {
-        if (origTrack.parameters() == trackParameters[idx].parameters()) {
-          particleAtVtx.insert(particleAtVtx.end(), particle);
+      for (std::size_t i = 0; i < trackParameters.size(); ++i) {
+        const auto& particle = associatedTruthParticles[i];
```

Comment:
Worth adding to the comment in line 340, that we assume that the associatedTruthParticles and the trackParameters vectors are aligned.